storage/kvstore.go

```diff
@@ -170,6 +171,44 @@ func (s *store) TxnDeleteRange(txnID int64, key, end []byte) (n, rev int64, err
 	return n, rev, nil
 }
 
+func (s *store) RangeEvents(key, end []byte, limit, startRev, endRev int64) (evs []storagepb.Event, nextRev int64, err error) {
+	if startRev <= s.compactMainRev {
+		return nil, 0, ErrCompacted
+	}
+
+	revs := s.kvindex.RangeEvents(key, end, startRev)
+	if len(revs) == 0 {
+		return nil, s.currentRev.main + 1, nil
```

Comment:
this seems to be racy. can currentRev be changed after the rangeEvent call finished?